pkg/addons/addons.go

```diff
@@ -57,9 +62,23 @@ func readDB(cs *acsapi.ContainerService, c *config.Config) (map[string]unstructu
 			return nil, err
 		}
 
+		azs, err := GetStorageAccountsClient(c)
+		if err != nil {
+			return nil, err
+		}
+
+		response, err := azs.ListKeys(context.Background(), c.ResourceGroup, c.RegistryStorageAccount)
+		if err != nil {
+			return nil, err
+		}
+
+		ex := &extra{
+			StorageAccountKey: *(((*response.Keys)[0]).Value),
+		}
+
 		ts := Translations[KeyFunc(o.GroupVersionKind().GroupKind(), o.GetNamespace(), o.GetName())]
 		for _, tr := range ts {
-			b, err := util.Template(tr.Template, nil, cs, c, nil)
+			b, err := util.Template(tr.Template, nil, cs, c, ex)
```

Comment:
Would it not be better to set it as c *Config object, so it will get persisted together with all config?